Add amcrest camera services and deprecate switches

pnbruckner opened the pull request:
## Description:
- Implement enabling and disabling motion detection from camera platform.
- Add amcrest specific camera services for controlling audio stream, motion recording, continuous recording and camera color mode, as well as moving camera to PTZ preset and starting and stopping PTZ tour function.
- Add camera attributes to indicate the state of the various camera settings controlled by the new services.
- Deprecate switches in favor of camera services and attributes.

#### Switch deprecation
The previously implemented switches are now redundant with the new camera services and attributes. Also, when the switches were first added, although merged, there was a comment afterwards that indicated they should probably have not been added, but rather the functionality should have been part of the camera platform. (See comment by @balloob at the end of PR #12992.) Therefore, for these reasons, the switches are being deprecated.

If a user would still like to use switches in the frontend they can add template switches that perform the same functions as the original switches by having them use the new camera services and attributes.

**Related issue (if applicable):**
N/A

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#9180

## Example entry for `configuration.yaml` (if applicable):
No change

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly ([example][ex-manifest]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).

[ex-manifest]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json
[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mobile_app/manifest.json#L5
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23
[manifest-docs]: https://developers.home-assistant.io/docs/en/development_checklist.html#_the-manifest-file_

pnbruckner committed:
Add amcrest camera services and deprecate switches

- Implement enabling and disabling motion detection from camera platform.
- Add amcrest specific camera services for controlling audio stream, motion recording, continuous recording and camera color mode, as well as moving camera to PTZ preset and starting and stopping PTZ tour function.
- Add camera attributes to indicate the state of the various camera settings controlled by the new services.
- Deprecate switches in favor of camera services and attributes.

pnbruckner commented:
I have a question about services. When I first wrote the code that is the basis of this PR, it was in the context of a custom version of the amcrest component & its platform code before the great migration. I only added services for the amcrest camera platform (i.e., camera/amcrest.py.) Therefore, following the instructions at the time, the services were registered under the 'camera' domain with names starting with the platform name (i.e., amcrest_xxx.)

Now, of course, that code has moved (with all the other amcrest platform code) to be under components/amcrest/camera.py. When I realized I had forgotten to write services.yaml to document the new services, I realized it would be problematic at best for the services to still be registered under the 'camera' domain.

So, I guess the questions are, 1) do I change the services from `camera.amcrest_xxx` to `amcrest.xxx`, or maybe even `amcrest.camera_xxx`; and 2) if I leave them as they were (in the 'camera' domain), how do I document them (in a services.yaml)?

pnbruckner commented:
I'm going to go ahead and change the services from `camera.amcrest_xxx` to `amcrest.xxx`, and move the service related code from `amcrest\camera.py` to `amcrest\__init__.py`, as well as do some other related reorganization. If someone reads this before I'm done and has a different opinion on the direction I should be going, by all means, please let me know.

pnbruckner committed:
Rename services and move service handling to __init__.py

Rename services from 'camera.amcrest_xxx' to 'amcrest.xxx'. This allows services to be documented in services.yaml.

Add services.yaml.

Reorganize hass.data[DATA_AMCREST] and do some general cleanup to make various platform modules more consistent.

Move service handling code to __init__.py from camera.py.

pnbruckner commented:
@amelchio @MartinHjelmare 
Hi guys. I tagged you since you helped me with some of my previous amcrest PRs.

Some background. I wrote a custom version of this component about a year ago that I and others have been using. It fixed some issues and added a lot of features. I've been slowing folding those changes into the standard component. The ultimate goal is to retire my custom component so that we can all go back to using the standard component.

Although I do have other enhancements that I'm already working on, and others still planned, this PR represents the last change needed so I can retire my custom component. So obviously the sooner it can be merged (with review comments, if any, addressed, of course), and make a release, the better.

Thanks for your consideration!

MartinHjelmare commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -139,22 +167,32 @@ def is_on(self):
 
+    async def async_added_to_hass(self):
+        """Add camera to list."""
+        self.hass.data[DATA_AMCREST]['cameras'].append(self)
```
We're not allowed to store entities in shared containers. Why do we need to store it?

MartinHjelmare replied:
The use case are the service handlers. Use our dispatch helper instead. Connect the signal here to the correct entity methods.

balloob replied:
@MartinHjelmare we should probably open a discussion if we should keep enforcing this, or that it's up to the integrations author if they want a decoupled design.

MartinHjelmare replied:
I've been thinking a bit about that. But so far I've only seen potential problems when not using dispatcher.

It's good to keep the platform contained and not expose it. Refactoring is easier. It's the same argument why we shouldn't interact directly with entities in tests but use the core to set up and assert.

It's also easier to introduce a bug and call a method before the entity has been added, when not using dispatcher, which often isn't allowed. With the dispatcher we've always follow the same pattern and use `async_added_to_hass`.

MartinHjelmare commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -170,18 +208,183 @@ def turn_on(self):
+    @callback
+    def async_enable_recording(self):
+        """Call the job and enable recording."""
+        return self.hass.async_add_job(self._enable_recording, True)
```
Use `self.hass.async_add_executor_job` instead.

pnbruckner replied:
I assume you mean for all instances. Will do.

MartinHjelmare commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -170,18 +208,183 @@ def turn_on(self):
+    # Additional Amcrest Camera service methods
+
+    @callback
+    def async_enable_recording(self):
```
Why are these callbacks? They should be coroutines since we want to await the task from the added job.

pnbruckner replied:
For this and the previous comment I was just mirroring what was being done in `homeassistant/components/camera/__init__.py` assuming that was correct.

Would it be correct this way?
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        return self.hass.async_add_executor_job(self._enable_recording, True)
```
Or does it need to be? (Sorry, still not 100% understanding asyncio.)
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        return await self.hass.async_add_executor_job(self._enable_recording, True)
```

balloob replied:
the latter. Camera doing a callback with return async_add_job was a "shortcut" that messes with stacktraces and we're trying to undo.

pnbruckner replied:
Actually I think it should be:
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        await self.hass.async_add_executor_job(self._enable_recording, True)
```
because the "normal" (i.e., non-coroutine) function ( `self._enable_recording` in this case) doesn't actually return anything. The return was only there because the callback was originally returning the task to be waited on. But now that is no longer the case, right?

The bigger picture is (and I'll use one of the other examples that is more straightforward) this coroutine (in `amcrest/__init__.py`):
```python
    async def async_goto_preset(service):
        preset = service.data[ATTR_PRESET]
        for camera in await async_extract_from_service(service):
            await camera.async_goto_preset(preset)
```
is calling (in `amcrest/camera.py`):
```python
    async def async_goto_preset(self, preset):
        """Call the job and move camera to preset position."""
        await self.hass.async_add_executor_job(self._goto_preset, preset)
```
So you can see there's nothing to return. Also, now I'm if there should even _be_ an `await` in the second function??? I would guess not and it really should be this:
```python
    async def async_goto_preset(self, preset):
        """Call the job and move camera to preset position."""
        self.hass.async_add_executor_job(self._goto_preset, preset)
```

MartinHjelmare commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -170,18 +208,183 @@ def turn_on(self):
+        from amcrest import AmcrestError
+
+        try:
+            self._api.audio_enabled = enable
```
This takes a boolean but not `_api.motion_detection`?

pnbruckner replied:
The camera API is bad. That became obvious in previous PRs. It uses properties and setters when they should be methods. I will hopefully change that some day. But for now, `self._api.audio_enabled` is a setter that will actually send a command to the camera via the network, and the value sent depends on the value being written to the setter, in this case `enable`. But, yes, I agree, given how properties and setters _should_ be used, this is confusing, but I can't fix it without making a huge breaking change to the amcrest pypi.org package.

MartinHjelmare commented on homeassistant/components/amcrest/sensor.py:
```diff
@@ -68,22 +64,22 @@ def icon(self):
         if self._sensor_type == 'motion_detector':
-            self._state = self._camera.is_motion_detected
-            self._attrs['Record Mode'] = self._camera.record_mode
+            self._state = self._api.is_motion_detected
```
Isn't this a binary state?

pnbruckner replied:
This sensor has been deprecated and there is already a binary_sensor taking its place (for just that reason. See PR #22703.) The change here actually has nothing to do with that. I've simply renamed `self._camera` to `self._api` throughout because I believe that is clearer.

MartinHjelmare commented on homeassistant/components/amcrest/sensor.py:
```diff
@@ -68,22 +64,22 @@ def icon(self):
-            self._state = self._camera.is_motion_detected
-            self._attrs['Record Mode'] = self._camera.record_mode
+            self._state = self._api.is_motion_detected
+            self._attrs['Record Mode'] = self._api.record_mode
```
State attribute keys should be lowercase snakecase.

pnbruckner replied:
Yes, planned for another PR. In fact, this whole sensor type will be removed, which will itself already be a breaking change. Trying to avoid this PR being a breaking change if at all possible.

MartinHjelmare replied:
Aren't we moving services already, ie a breaking change?

pnbruckner replied:
No, adding services that never existed. (When I originally wrote them almost a year ago as part of my custom component version it was before the great migration and the services were added to `camera/amcrest.py`. The "move" is happening as part of me trying to fold those new features into the standard component and realizing they now probably belonged in `amcrest/__init__.py` instead of `amcrest/camera.py`. Does that make sense?)

MartinHjelmare replied:
Right. If there's no breaking changes in this PR, it's ok to wait with those changes to another PR, and then try to group all of them together.

MartinHjelmare commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+    async def async_extract_from_service(service):
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids == ENTITY_MATCH_ALL:
+            return [entity for entity in hass.data[DATA_AMCREST]['cameras']
```
We should use our dispatch helper to signal the correct entity instead of storing the entity in a shared container. Entities are a platform concern.

pnbruckner replied:
Sorry, but I'm not sure I'm following you. This is an amcrest service, not a camera service. And it doesn't apply to all cameras, just amcrest cameras. How could I use the dispatcher helper to do figure out all the amcrest cameras that exist?

MartinHjelmare replied:
In `async_added_to_hass` connect a armcrest unique signal per service to the corresponding entity method.

pnbruckner replied:
Yes, that makes sense. I've seen the dispatcher methods before. I just didn't think they applied here. Actually I didn't even think of using them because, again, the component was already using `hass.data` before I came on the scene. 😄 I can work on changing this as you suggest.

pnbruckner replied:
I started looking into making this change, looking at existing example uses of the dispatcher mechanism, and it became clear that I totally misunderstood what you were asking me to do.

It seems you're asking me to no longer keep a list of amcrest camera entities at all. And, hence, you don't want the "hub" service handler to call the camera's (or cameras') service method directly. But rather the camera should connect itself to one signal per service (so a total of ten per camera.) And then the hub should just use the dispatcher to send the service's corresponding signal (with call args if necessary) to "wake up" the camera(s) to run the corresponding service method. Am I even close to understanding now?

If so, I'm not seeing how that will work.

How does the hub "signal the correct entity"? Doesn't it just send a signal, and wouldn't that cause all the cameras that are "listening" on that signal to respond and call the service method? How would they know whether or not they were the "correct" entity, or one of the "correct" entities? Where does the "async_extract_from_service" processing happen? It can't happen in the hub anymore, because it doesn't have the list of amcrest cameras anymore. And how can it happen in the camera if it doesn't have access to the call information? Or do I need to pass the call information along with the dispatcher signal and other possible args? And, if so, does that mean all the camera entities have to individually run the equivalent "async_extract_from_service" processing? Seems inefficient.

Can you point me to a specific, existing use of the dispatcher that is equivalent to what you want me to do here? I found a lot of examples but I didn't seem to find any like this.

pnbruckner replied:
@MartinHjelmare, after giving it a lot more thought, and looking at more examples, I think I now understand what I need to do. I'll be pushing two more commits. The first will include the rebase and all the other requested changes, and the second will implement the use of the dispatcher. Hopefully it's what you're looking for! :smile:

pnbruckner commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -71,70 +76,110 @@
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            cv.time_period,
+        vol.Optional(CONF_BINARY_SENSORS):
+            vol.All(cv.ensure_list_csv, [vol.In(BINARY_SENSORS)]),
```
Sorry, I don't understand that. I think it is very helpful and see no downside. Is this a decision the team has made? Can you explain the reasoning? If there's a good reason, by all means I'll change it. I just don't understand why it's bad.

MartinHjelmare replied:
Our config interface should be kept as consistent as possible. We use yaml lists for lists of things. Json lists are compatible with that.

We shouldn't introduce variability that we don't really need. Keep it simple. Simple is predictable.

pnbruckner replied:
Okey doke. Thanks. Will do.

pnbruckner commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+        for camera in await async_extract_from_service(service):
+            await camera.async_set_color_bw(cbw)
+
+    handler_services = {
```
Except there are multiple services, not multiple handlers, so it would then be misleading. How about `handled_services`?

pnbruckner replied:
I changed it to `handled_services`.

pnbruckner commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+        SERVICE_STOP_TOUR: 'async_stop_tour',
+    }
+
+    if not hass.services.has_service(DOMAIN, SERVICE_ENABLE_RECORDING):
```
Ok, thanks for confirming that. I wasn't sure if it would ever get called multiple times. I'll change it.

pnbruckner commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -5,45 +5,52 @@
     """An implementation of an Amcrest IP camera."""
 
-    def __init__(self, hass, amcrest):
+    def __init__(self, name, device, hass):
```
Ok. That's the way it was before I came on the scene, but I can do that.

pnbruckner commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -105,6 +112,22 @@ def name(self):
+        """Return the Amcrest-spectific camera state attributes."""
+        attr = {}
+        if self.motion_detection_enabled is not None:
+            attr['motion_detection'] = _BOOL_TO_STATE.get(
```
Yes, but the base doesn't add it to the attributes correctly, IMHO. It only adds it if it is True, not if it is False. (I.e., it only does `if self.motion_detection_enabled`, not `if self.motion_detection_enabled is not None`.) I was trying to work around that flaw.

balloob replied:
It's not up to integrations to fix things in the base

pnbruckner replied:
Ok, fair enough. I'll just take this out.

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -5,19 +5,24 @@
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.service import async_extract_entity_ids
 
 REQUIREMENTS = ['amcrest==1.3.0']
```
Your branch still having this means it's quite outdated, please rebase.

pnbruckner replied:
Yes, saw that. I was wondering if I should remove them, and I saw they already were. This is only that old because it sat around for a week. 😄 I'll rebase again. Thanks!

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -71,70 +76,110 @@
+                                device[CONF_PORT],
+                                username,
+                                password).camera
             # pylint: disable=pointless-statement
```
Please add a comment why this call is not pointless

pnbruckner replied:
Yes, I considered doing that. (This component, and the pypi.org package it uses, have been around for a while. I'm pretty late to the game. Trying not to solve all the world's problems in one PR. 😄 ) But I can certainly do that.

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
 
-        hass.data[DATA_AMCREST][name] = AmcrestDevice(
-            camera, name, authentication, ffmpeg_arguments, stream_source,
+        hass.data[DATA_AMCREST]['devices'][name] = AmcrestDevice(
```
Are names always unique?

pnbruckner replied:
Yes, there's a custom validator to make sure. See [here](https://github.com/pnbruckner/home-assistant/blob/183a07345715e92f5bedd223ba74111b8e042631/homeassistant/components/amcrest/__init__.py#L99-L102) and [here](https://github.com/pnbruckner/home-assistant/blob/183a07345715e92f5bedd223ba74111b8e042631/homeassistant/components/amcrest/__init__.py#L133-L135).

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+
+    async def async_service_handler(service):
+        for camera in await async_extract_from_service(service):
+            await getattr(camera, handler_services[service.service])()
```
Make sure you're checking permissions.

pnbruckner replied:
That's on the list of things to do. Again, not trying to do everything at once. I'm trying to enhance this component slowly but surely, not rewrite it in one huge undertaking.

balloob replied:
leave services to the next PR ?

pnbruckner replied:
The services are pretty much the whole point of this PR.

pnbruckner replied:
Ok, permission checking added (which will be in next commit.)

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+    async def async_goto_preset(service):
+        preset = service.data[ATTR_PRESET]
+        for camera in await async_extract_from_service(service):
+            await camera.async_goto_preset(preset)
```
Make sure you're checking permissions.

pnbruckner replied:
Same as above.

balloob commented on homeassistant/components/amcrest/__init__.py:
```diff
@@ -160,47 +205,94 @@ def setup(hass, config):
+    async def async_set_color_bw(service):
+        cbw = service.data[ATTR_COLOR_BW]
+        for camera in await async_extract_from_service(service):
+            await camera.async_set_color_bw(cbw)
```
Make sure you're checking permissions.

pnbruckner replied:
Same as above.

pnbruckner commented:
@MartinHjelmare @balloob 
Thanks for taking the time to review! I'll work on the requested changes and make a new commit hopefully in the next couple days.

pnbruckner committed:
Update per review comments, part 1

- Rebase
- Add permission checking to services
- Change cv.ensure_list_csv to cv.ensure_list
- Add comment for "pointless-statement" in setup
- Change handler_services to handled_services
- Remove check if services have alreaday been registered
- Pass ffmpeg instead of hass to AmcrestCam __init__
- Remove writing motion_detection attr from device_state_attributes
- Change service methods from callbacks to coroutines

pnbruckner committed:
Update per review comments, part 2

- Use dispatcher to signal camera entities to run services.
- Reorganize a bit, including moving a few things to new modules const.py & helpers.py.

MartinHjelmare commented on homeassistant/components/amcrest/camera.py:
```diff
@@ -168,18 +265,177 @@ def turn_on(self):
+
+    # Additional Amcrest Camera service methods
+
+    async def async_enable_recording(self, call_data):
```
It's probably better if the entity doesn't need to know about the call data. This method should just take whatever info it needs.

pnbruckner replied:
The knowledge of the service, from the schema to the method that implements it, needs to be somewhere.

Originally it was all in `camera.py`. Then I moved much of it to `__init__.py`, leaving just the camera methods, with no service knowledge, in `camera.py`. But that required the "hub" code to have too much knowledge (and direct access to) the camera entities.

So then you had me move much of the knowledge back to `camera.py`, leaving just the service registration and processing in `__init__.py`, and I did that in such a way as to decouple the two as much as possible. That means most of the camera specific knowledge is now in `camera.py`, including the service schema and the extraction of the service data. To me, that makes the most sense, since it's all camera specific, and keeps it all in one place.

But now we still have the service registration and initial service handling, including the permission checking, in `__init__.py`, with everything else in `camera.py.` And what is the benefit? I don't see it. In fact, it seems to do nothing but add overhead, specifically the dispatching mechanism. Now when a service is called, not only is there the normal service processing, there's also an extra layer of dispatching, with the dispatcher being clogged up with many more events that don't seem to add any benefit, just extra processing and delay.

At this point I would propose moving the service registration and handling back to `camera.py` so everything is in one place. And I could go back to the way it was originally where the handler extracted the data from the call, which is what you're basically asking for here. The extraction of the entity ids and permission checking can go to the new `helpers.py` module. (In fact, if the code in helpers.service.entity_service_call were refactored, taking an entity id list instead of platforms, I could use it directly and not have to mostly duplicate it.)

pnbruckner replied:
Ok, done.

pnbruckner committed:
Update per review comments, part 3

Move call data extraction from camera.py to __init__.py.

MartinHjelmare reviewed (approved):
The parts I've commented on looks good! @balloob should look at the service permissions.

JLFN commented:
Please add so i  choose channel also its default channel 0.
snapshot.py
.py``` # -*- coding: utf-8 -*-
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, version 2 of the License.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# vim:sw=4:ts=4:et
import shutil


class Snapshot(object):
    def __get_config(self, config_name):
        ret = self.command(
            'configManager.cgi?action=getConfig&name={0}'.format(config_name)
        )
        return ret.content.decode('utf-8')

    @property
    def snapshot_config(self):
        return self.__get_config('Snap')

    def snapshot(self, channel=0, path_file=None, timeout=None):
        """
        Args:

            channel:
                Values according with Amcrest API:
                0 - regular snapshot
                1 - motion detection snapshot
                2 - alarm snapshot

                If no channel param is used, default is 0

            path_file:
                If path_file is provided, save the snapshot
                in the path

        Return:
            raw from http request
        """
        ret = self.command(
            "snapshot.cgi?channel={0}".format(channel),
            timeout_cmd=timeout
        )

        if path_file:
            with open(path_file, 'wb') as out_file:
                shutil.copyfileobj(ret.raw, out_file)

        return ret.raw
```

pnbruckner commented:
@JLFN the comments in snapshot.py are (and have been) misleading regarding the channel parameter. It actually selects from the video streams. 0 is the Main Stream, and 1 is the Sub Stream. In HA it is controlled by the [resolution configuration variable](https://www.home-assistant.io/components/amcrest/#resolution). The value of `high` corresponds to channel 0, and the value of `low` corresponds to channel 1. So, if you need channel 1, then you should set `resolution` to `low` in your amcrest configuration. And, BTW, this is probably not the best place to make a feature request.

pnbruckner commented:
@balloob, per @MartinHjelmare's comment above, do the service permissions look ok to you? Is this PR ready to be merged? Thanks!

pnbruckner commented:
@MartinHjelmare, Paul is a busy man and this probably fell through the cracks. Is there any way you or someone else can check the service permissions? I think that's the only thing pending on this PR. Thanks!

MartinHjelmare commented:
Any member that feels comfortable can check the service permissions.

balloob commented:
Looks good!

akerrichardt commented:
I have several Amcrest camera’s working with home assistants. Two of these camera’s use the switch option activated by motion detection from a binary RF motion detection unit. Everything worked until I upgraded to 0.93.1. I have removed all the Amcrest Swithch options in my config.yaml and changed the automatons.from switch.turn_on to the Service enable_motion_recording / Service enable_motion_recording/disable_motion_recording. Motion at my motion detector turns on the record motion option in my amcrest camera but the the Service enable_motion_recording / disable_motion_recording does not turn off the cameras record function.
Any help would be appreciated. Thanks

pnbruckner commented:
@akerrichardt I think you should be opening an issue instead of commenting on the PR. Please open an issue and tag me. I'll respond there. Thanks.
